code/modules/mob/living/carbon/human/species/working_joe/_species.dm

```diff
@@ -0,0 +1,129 @@
+/datum/species/synthetic/colonial/working_joe
+	name = SYNTH_WORKING_JOE
+	name_plural = "Working Joes"
+	uses_ethnicity = FALSE
+	mob_inherent_traits = list(TRAIT_SUPER_STRONG, TRAIT_INTENT_EYES, TRAIT_EMOTE_CD_EXEMPT)
+
+	hair_color = "#000000"
+	icobase = 'icons/mob/humans/species/r_synthetic.dmi'
+	deform = 'icons/mob/humans/species/r_synthetic.dmi'
+
+
+/datum/species/synthetic/colonial/working_joe/handle_post_spawn(mob/living/carbon/human/joe)
+	. = ..()
+	give_action(joe, /datum/action/joe_emote_panel)
+
+
+/// Open the WJ's emote panel, which allows them to use voicelines
+/datum/species/synthetic/colonial/working_joe/proc/open_emote_panel()
+	var/datum/joe_emote_panel/ui = new(usr)
```

Comment:
if the panel is new'd on every action, and the anti spam cooldown is on the panel, people could just press the action newly every time. but i guess all that effort probably takes just as much time

Reply:
I did notice that, but i'm not very concerned because:
- As you said, it'll take about as long to reopen the panel and select the correct category as the short 2.5s cooldown is
- the role is whitelisted and is expected to not abuse the toys they're given
- you could already do this by copy-paste spamming an emote, since wjs inherently have no emote cd

Ultimately, the cooldown's less so meant as a grief protection tool, and more to prevent "whoops i double clicked and now the voice lines are overlapping"

Reply:
i forgot about 2) lol